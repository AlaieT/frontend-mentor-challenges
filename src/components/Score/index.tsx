import React from "react";

import type { ScoreProps } from "../../types";

import styles from "../../styles/components/score.module.scss";

const Score = ({ score, delay }: ScoreProps) => {
  const [count, setCount] = React.useState(0);

  React.useEffect(() => {
    const interval = setInterval(() => {
      setCount((item) => {
        if (item === score) {
          clearInterval(interval);
          return item;
        }
        return item + 1;
      });
    }, delay);
    return () => clearInterval(interval);
  }, [score, delay]);

  return (
    <div id={styles.score}>
      <p>
        <span>{count}</span>&nbsp;&#8212;&nbsp;completed 
      </p>
    </div>
  );
};

export default Score;
