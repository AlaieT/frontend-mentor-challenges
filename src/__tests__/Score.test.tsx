import React from "react";
import { render, cleanup, act } from "@testing-library/react";

import Score from "../components/Score";

afterEach(cleanup);

describe("Score", () => {
  describe("correct render", () => {
    it("should match snapshot before animation ends", () => {
      expect(
        render(<Score score={100} delay={25} />).asFragment()
      ).toMatchSnapshot();
    });

    it("should match snapshot after animation ends", async () => {
      const { asFragment } = render(<Score score={10} delay={25} />);

      await act(async () => {
        await new Promise((resolve, _) =>
          setTimeout(() => resolve("done"), 275)
        );
      });

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
