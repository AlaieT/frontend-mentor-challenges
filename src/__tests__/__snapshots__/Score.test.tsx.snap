// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score correct render should match snapshot after animation ends 1`] = `
<DocumentFragment>
  <div
    id="score"
  >
    <p>
      <span>
        10
      </span>
       — completed
    </p>
  </div>
</DocumentFragment>
`;

exports[`Score correct render should match snapshot before animation ends 1`] = `
<DocumentFragment>
  <div
    id="score"
  >
    <p>
      <span>
        0
      </span>
       — completed
    </p>
  </div>
</DocumentFragment>
`;
